src/app/common/validation/forms/address-validators.ts

```diff
@@ -20,27 +20,12 @@ export function notCurrentAddressValidator(currentAddress: string) {
 }
 
 export function btcAddressValidator() {
-  return yup
-    .string()
-    .defined(FormErrorMessages.AddressRequired)
-    .test((input, context) => {
-      if (!input) return false;
-      if (!validate(input))
-        return context.createError({
-          message: FormErrorMessages.InvalidAddress,
-        });
-      return true;
-    });
-}
-
-// ts-unused-exports:disable-next-line
-export function btcTaprootAddressValidator() {
-  return yup.string().test((input, context) => {
-    if (!input || !validate(input)) return false;
-    if (getAddressInfo(input).type !== AddressType.p2tr)
+  return yup.string().test((value, context) => {
+    if (isString(value) && value !== '' && !validate(value)) {
```

Comment:
Maybe we can extend `isString` so that it accepts as prop to check for non-empty and we don't need to repeat `value !== ''`

Reply:
maybe new `isNonEmptyString` helper?

Reply:
Good idea, but I wonder if it would break use somewhere if expecting to be ok with an empty string? Prob a good idea to add a shared function though for `isStringAndNotEmpty`? What do you think?

Reply:
Sounds good 👍 not an urgent thing to do, just mentioning along the way

Reply:
I must not have refreshed to see @alter-eggo suggestion here before my reply. 👍

Reply:
https://github.com/leather-io/mono/pull/266